From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SoSeDiK <mrsosedik@gmail.com>
Date: Thu, 16 May 2024 11:37:52 +0300
Subject: [PATCH] Parse items in show_item hover event

Allows parsing items inside hover event. Required to not kick the client when an injected item is shown.
By default only parses hovers in system chat messages, but can be enabled globally via "item-modifiers.apply-modifiers-on-all-hover" option.

diff --git a/src/main/java/me/sosedik/kiterino/modifier/item/context/ItemModifierContextType.java b/src/main/java/me/sosedik/kiterino/modifier/item/context/ItemModifierContextType.java
index 3b31356890f1843de7a0fabec64342ac5cc49a16..15218759093c081537a7af92b8f8aa5324701c9a 100644
--- a/src/main/java/me/sosedik/kiterino/modifier/item/context/ItemModifierContextType.java
+++ b/src/main/java/me/sosedik/kiterino/modifier/item/context/ItemModifierContextType.java
@@ -63,6 +63,11 @@ public class ItemModifierContextType {
 	 */
 	public static final ItemModifierContextType EMPTY_NO_LORE = context(ItemModifierContext.EMPTY.getClass()).withName().build();
 
+	/**
+	 * Show item hover event in text components
+	 */
+	public static final ItemModifierContextType CHAT_ITEM = context(ItemModifierContext.EMPTY.getClass()).withName().withLore().build();
+
 	private final Class<? extends ItemModifierContext> contextClass;
 	private final boolean visibleName;
 	private final boolean visibleLore;
